src/main/scala/mesosphere/marathon/core/storage/impl/BasePersistenceStore.scala

```diff
@@ -0,0 +1,145 @@
+package mesosphere.marathon.core.storage.impl
+
+import java.time.OffsetDateTime
+
+import akka.http.scaladsl.marshalling.{ Marshal, Marshaller }
+import akka.http.scaladsl.unmarshalling.{ Unmarshal, Unmarshaller }
+import akka.stream.Materializer
+import akka.stream.scaladsl.{ Keep, Sink, Source }
+import akka.{ Done, NotUsed }
+import mesosphere.marathon.core.storage.PersistenceStore
+import mesosphere.util.LockManager
+
+import scala.async.Async.{ async, await }
+import scala.concurrent.{ ExecutionContext, Future }
+
+/**
+  * Persistence Store that handles all marshalling and unmarshalling, allowing
+  * subclasses to focus on the raw formatted data.
+  *
+  * @tparam K The persistence store's primary key type
+  * @tparam Serialized The serialized format for the persistence store.
+  */
+abstract class BasePersistenceStore[K, Category, Serialized](implicit
+  ctx: ExecutionContext,
+  mat: Materializer) extends PersistenceStore[K, Category, Serialized]
+    with TimedPersistenceStore[K, Category, Serialized] {
```

Comment:
That was the intent yes. Operations at the lead persistence layer should be
timed. I'd like yo eventually add error tracking as well.

On Tue, Jul 12, 2016 at 9:34 AM Ken Sipe <notifications@github.com> wrote:

> In
> src/main/scala/mesosphere/marathon/core/storage/impl/BasePersistenceStore.scala
> <https://github.com/mesosphere/marathon/pull/4062#discussion_r70473214>:
>
> > +import mesosphere.util.LockManager
> > +
> > +import scala.async.Async.{ async, await }
> > +import scala.concurrent.{ ExecutionContext, Future }
> > +
> > +/**
> > +  * Persistence Store that handles all marshalling and unmarshalling, allowing
> > +  * subclasses to focus on the raw formatted data.
> > +  *
> > +  * @tparam K The persistence store's primary key type
> > +  * @tparam Serialized The serialized format for the persistence store.
> > +  */
> > +abstract class BasePersistenceStore[K, Category, Serialized](implicit
> > +  ctx: ExecutionContext,
> > +  mat: Materializer) extends PersistenceStore[K, Category, Serialized]
> > +    with TimedPersistenceStore[K, Category, Serialized] {
>
> Metrics is always present then? or should this be configurable?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4062/files/5ffaa7e4c7657fc609814fd932a7343acfb909f0#r70473214>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAweZlHiYqoAGlEZOkDgL0DVqdqY72uMks5qU8H5gaJpZM4JDjhV>
> .
>